src/main/java/sg/ncl/DataController.java

```diff
@@ -0,0 +1,236 @@
+                                         Model model, @PathVariable Optional<String> id,
+                                         HttpSession session) throws WebServiceRuntimeException {
+        if (bindingResult.hasErrors()) {
+            StringBuilder message = new StringBuilder();
+            message.append("Error(s):");
+            message.append("<ul class=\"fa-ul\">");
+            for (ObjectError objectError : bindingResult.getAllErrors()) {
+                FieldError fieldError = (FieldError) objectError;
+                message.append("<li><i class=\"fa fa-exclamation-circle\"></i> ");
+                message.append(fieldError.getField());
+                message.append(" ");
+                message.append(fieldError.getDefaultMessage());
+                message.append("</li>");
+            }
+            message.append("</ul>");
+            model.addAttribute("message", message.toString());
+            return CONTRIBUTE_DATA_PAGE;
+        }
+
+        JSONObject dataObject = new JSONObject();
+        dataObject.put("name", dataset.getName());
+        dataObject.put("description", dataset.getDescription());
+        dataObject.put("contributorId", session.getAttribute("id").toString());
+        dataObject.put("visibility", dataset.getVisibility());
+        dataObject.put("accessibility", dataset.getAccessibility());
+        dataObject.put("resources", new ArrayList());
+        dataObject.put("approvedUsers", new ArrayList());
+        dataObject.put("releasedDate", dataset.getReleasedDate());
+        log.debug("DataObject: {}", dataObject.toString());
+
+        HttpEntity<String> request = createHttpEntityWithBody(dataObject.toString());
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+
+        ResponseEntity response = getResponseEntity(id, request);
+        String dataResponseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(dataResponseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+                switch (exceptionState) {
+                    case DATASET_NAME_IN_USE_EXCEPTION:
+                        log.warn("Dataset name already exists.");
+                        model.addAttribute("message", "Error(s):<ul><li>dataset name already exists</li></ul>");
+                        break;
+                    case FORBIDDEN_EXCEPTION:
+                        log.warn("Saving of dataset forbidden.");
+                        model.addAttribute("message", "Error(s):<ul><li>saving dataset forbidden</li></ul>");
+                        break;
+                    default:
+                        log.warn("Unknown error.");
+                }
+
+                return CONTRIBUTE_DATA_PAGE;
+            }
+        } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)